ingestion/src/main/java/feast/ingestion/transform/fn/ValidateFeatureRowDoFn.java

```diff
@@ -57,15 +61,12 @@ public void processElement(ProcessContext context) {
     String error = null;
     FeatureRow featureRow = context.element();
     FeatureSet featureSet = getFeatureSets().getOrDefault(featureRow.getFeatureSet(), null);
+    Set<FieldProto.Field> fields = new HashSet<>();
     if (featureSet != null) {
-
       for (FieldProto.Field field : featureRow.getFieldsList()) {
         Field fieldSpec = featureSet.getField(field.getName());
         if (fieldSpec == null) {
-          error =
-              String.format(
-                  "FeatureRow contains field '%s' which do not exists in FeatureSet '%s' version '%d'. Please check the FeatureRow data.",
-                  field.getName(), featureSet.getReference());
+          // skip
```

Comment:
What if we logged statistics in terms of unnecessary fields so that it doesnt result an an actual error?

Reply:
Possible, but is it possible if i opened a separate PR for this? It would involve introducing a new TupleTag (for warnings)

Reply:
Can't we use our metrics client here?

Reply:
we need to bubble the error down to the metrics writing fn